### DISCUSSION TITLE
Class docstring applied to fields of class (as Pycharm does)

### DISCUSSION THREAD

**Discussion by xxc-zsz** [microsoft/pylance-release#4759] [Enhancement]
**Code:**

```
@dataclass
class SessionMetadata:
    """Metadata of a Session

    Attributes:
        created_at: Date of creation, in POSIX timestamp format.
        usage_count: The number of times the session was applied to the payload.
    """

    created_at: int
    usage_count: int
```

**Vscode:**

![image](https://github.com/microsoft/vscode-python/assets/108051416/c51d5a06-70f4-49f2-af60-cf54fa65bd6e)

**PyCharm:**

![image](https://github.com/microsoft/vscode-python/assets/108051416/62e6c62d-0e38-480f-ad49-1246244d3e69)

_If I write the code like this in VScode, the hover document can be displayed:_

```
@dataclass
class SessionMetadata:
    created_at: int
    """
    Date of creation, in POSIX timestamp format.
    """

    usage_count: int
    """
    The number of times the session was applied to the payload.
    """
```

![image](https://github.com/microsoft/vscode-python/assets/108051416/c9a78f4f-8df8-4e8c-afef-0c07f9a38fbd)

**But how to display the document in the first way of annotation?**

**Comment by starball5** [comment]
Related on Stack Overflow: [In VSCode, how to get on-hover documentation shown for class properties documented in the class docstring?](https://stackoverflow.com/q/76876719/11107541)

Also, why is this (currently) phrased as a support question if it's a feature-request?

**Comment by clouedoc** [comment]
I can confirm this issue/lack of feature. In fact, I posted the original StackOverflow question 😃 
Thanks a lot for opening this issue, @xxc-zsz

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by prokie** [comment]
I would really like this feature to be added.

**Comment by mubarizafzal** [comment]
I would also really appreciate this feature being added!

**Comment by matsalleh92** [comment]
I would like this feature please!

**Comment by BryanFauble** [comment]
This would be very useful - as of now I have to duplicate these docstrings in both places: 1) for mkdocs, 2) for hoverdocs

**Comment by heddxh** [comment]
I would like this feature!

**Comment by robwoods-cam** [comment]
I would benefit from this feature!

**Comment by ChariseWalraven** [comment]
Same!

**Comment by JacobHast** [comment]
I would also love this!

**Comment by gomills** [comment]
I wouldn't complain as well! Looking forward for this feature. It bastly enhances performance as now I have to switch tabs constantly

**Comment by bstableynv5** [comment]
Yes, I would also like attribute documentation in the class docstring to appear when hovering over attributes. I think if, for some reason, a docstring appears on _both_ the attribute (at creation) and the class docstring, either (1) both are concatenated somehow or (2) the attribute's docstring takes precedence.

I recently had to "downgrade" my attribute docstrings for code review because of this. Pycharm apparently does show the attribute docstrings that are defined in the class docstring when hovering over members. I don't know if it supports docstrings on the attribute as vscode does.

**Comment by rchiodo** [comment]
Our next prerelease version of Pylance should have this feature added to it.
